### ISSUE TITLE
One is perpetually shown as online

### ISSUE THREAD

**Issue by adrianghc** [UnigramDev/Unigram#1117]
People report to me that I'm being perpetually shown as online since a few days. Since that's when I downloaded the beta release for the new desktop version of Unigram and since I'm the only person I know who uses it, I suspect that one of the newest versions of Unigram is indeed the culprit here. I'm using a Surface Pro with Modern Standby, so it couldn't the desktop version as I'm being shown as online when my Surface is in standby and the official desktop client of Telegram couldn't do that.

**Comment by MarcAnt01** [comment]
I think this is bug is due to your PC, as soon as I click on taskbar or open another app I am offline

**Comment by FrayxRulez** [comment]
Known

**Comment by NyanLeek** [comment]
It only happens when you close the client. You're briefly showed offline then back online even if the app stays closed. I want to receive notifications but be shown as offline if I don't have the app open, I think that's the intended behavior.

**Comment by adrianghc** [comment]
This still seems to be an issue.

**Comment by massijay** [comment]
> This still seems to be an issue.
For me too

**Comment by Jamesits** [comment]
I have observed this bug many times but I can't tell exactly how to reproduce it. I've been running Unigram on multiple computers simultaneously and it just happens randomly.

When this bug is happening, if I minimize Unigram, my account would be briefly shown as offline then online forever. Quitting the app _sometimes_ help, but in most cases I have to log out of the session to get it solved. 

I think this bug has been around since Unigram had tray icon support (and as an educated guess, may be related to the tray icon implementation). I'm currently using 3.10.2510.

**Comment by FrayxRulez** [comment]
I'd mark this as closed as the system tray thing is dead

**Comment by massijay** [comment]
I think this problem exists again

**Comment by MrFobwatch** [comment]
I too have been having this issue recently. I've had to stop using Unigram temporarily because of it.

**Comment by MahdiGhiasi** [comment]
I've experimented with killing UnigramBridge.exe manually while Unigram is open, and then closing the Unigram window, in that case my status stayed offline. But the next time I opened Unigram (and UnigramBridge.exe also opened by it), again I was shown as online after closing Unigram window.

Just wanted to confirm that this indeed seems to be related to UnigramBridge.exe.

On a side question, does UnigramBridge.exe do anything besides the tray icon? And if not, can it *not* start when Show Tray Icon setting is disabled, as a workaround for this?

**Comment by MarcAnt01** [comment]
> I've experimented with killing UnigramBridge.exe manually while Unigram is open, and then closing the Unigram window, in that case my status stayed offline. But the next time I opened Unigram (and UnigramBridge.exe also opened by it), again I was shown as online after closing Unigram window.
> 
> Just wanted to confirm that this indeed seems to be related to UnigramBridge.exe.
> 
> On a side question, does UnigramBridge.exe do anything besides the tray icon? And if not, can it _not_ start when Show Tray Icon setting is disabled, as a workaround for this?

It allows the app to run in background

**Comment by MahdiGhiasi** [comment]
I assumed the notifications still come from WNS, and not manually checked by a win32 process, and if that case there's no need for app to run on background. 

Isn't that true any longer?

@MarcAnt01 @FrayxRulez

**Comment by MarcAnt01** [comment]
> I assumed the notifications still come from WNS, and not manually checked by a win32 process, and if that case there's no need for app to run on background.
> 
> Isn't that true any longer?
> 
> @MarcAnt01 @FrayxRulez

https://github.com/UnigramDev/Unigram/issues/1206

**Comment by MarcAnt01** [comment]
In my iPhone Telegram uses a notification system similar to the one of Unigram (in-app notifications with the app open and push when it's closed) and I am quite disappointed, because if I have read the conversation with the notification through another device, it still appears in the notification centre and I have to clear the notifications manually. Moreover, if I have Unigram and my Android device with me, I notice that they get notifications faster. That's why I have manually disabled push notifications

**Comment by massijay** [comment]
From my experience the app worked way better before the Win32 bridge was introduced

**Comment by MarcAnt01** [comment]
> From my experience the app worked way better before the Win32 bridge was introduced

What do you mean by that?

**Comment by massijay** [comment]
> > From my experience the app worked way better before the Win32 bridge was introduced
> 
> What do you mean by that?

I never had this and notification (missing or double) problems before

**Comment by wobbleninja** [comment]
My friend told me the same thing one day. That I'm always online and I instantly figured out that Unigram would trigger this and I thought that I don't care and honestly I took it as a "feature".

But apparently now that I use Telegram more often for chatting, it seems that this is annoying since people think I am around and honestly I am actually around when I use Unigram since it means I am on my PC and I can instantly answer but don't get me started about "you are online and you don't read my messages" gf problems....................

Is this a bug or that's how supposed to be? Unless you kill the task from task manager or simply press "Quit Unigram" from tray icon you're always online.

**Comment by MarcAnt01** [comment]
> My friend told me the same thing one day. That I'm always online and I instantly figured out that Unigram would trigger this and I thought that I don't care and honestly I took it as a "feature".
> 
> But apparently now that I use Telegram more often for chatting, it seems that this is annoying since people think I am around and honestly I am actually around when I use Unigram since it means I am on my PC and I can instantly answer but don't get me started about "you are online and you don't read my messages" gf problems....................
> 
> Is this a bug or that's how supposed to be? Unless you kill the task from task manager or simply press "Quit Unigram" from tray icon you're always online.

![image](https://user-images.githubusercontent.com/43090263/79631225-02a9a980-8158-11ea-8321-613b5b771a5c.png)

**Comment by wobbleninja** [comment]
**TL:DR: Unigram properly takes you offline now when you close its window.**

Quick heads up just in case anyone come across to this thread or wants an update.

I updated Unigram today and it seems that the tray icon is gone (even though the advanced settings have the option to show the tray icon, and I have it enabled) although this means every time you hit 'X' to close Unigram it properly takes you offline now.

I have no problem having no tray icon, I wanted the tray icon to quickly "quit Unigram" so I can put myself offline manually and not killing the Task from Task Manager every time. Now, since you can just close Unigram window and you're offline properly, I think I'm fine with it.

**Comment by MarcAnt01** [comment]
Yes, closing it

**Comment by MarcAnt01** [comment]
Tray icon is nice, but having a bug like this, doesn't make it worthy
